issue_manager_script.py

```diff
@@ -127,114 +92,125 @@
 
-# If the project was not found, exit the script
 if not project:
     print(f"Project number {project_number} not found.")
     exit(1)
 
-# Edge Case 16: Prevent overlapping runs
-# Create a lock file to ensure only one instance runs at a time
-lock_file = '/tmp/why_drs_issue_manager.lock'
+# ----------------------- Prevent Overlapping Runs ----------------------------
 
+# Edge Case 16: Concurrency and Race Conditions
+# Implement a lock file mechanism to prevent overlapping runs
 if os.path.exists(lock_file):
-    print("Another instance of the script is already running. Exiting to prevent overlap.")
-    exit(1)
+    # Check if the lock file is older than 1 hour
+    if time.time() - os.path.getmtime(lock_file) > 3600:
+        os.remove(lock_file)  # Remove stale lock file
+    else:
+        print("Another instance of the script is already running. Exiting to prevent overlap.")
+        exit(1)
 else:
     open(lock_file, 'w').close()  # Create the lock file
 
+# -------------------------- Main Processing Loop -----------------------------
+
 try:
-    # Edge Case 2 and 12: Calculate the timestamp for 24 hours ago with a buffer
+    # Edge Case 2 and 12: Time Zone and Timestamp Issues
+    # Calculate the timestamp for 24 hours ago with a 5-minute buffer
     since_time = datetime.utcnow() - timedelta(days=1, minutes=5)  # 5-minute buffer
 
-    # Fetch all repositories within the organization with proper pagination
+    # Fetch all repositories within the organization
     try:
         repos = org.get_repos()
     except GithubException as e:
-        print(f"Failed to get repositories for organization {org_name}: {e}")
+        print(f"Failed to get repositories for organization {org_name}: {e.data}")
         exit(1)
 
     # Iterate over each repository
     for repo in repos:
-        # Edge Case 25: Handle repositories transferred out of the organization
+        # Edge Case 25: Changes in Repository Ownership
+        # Ensure the repository belongs to the organization
         if repo.organization.login.lower() != org_name.lower():
             continue  # Skip repositories not belonging to the organization
 
-        # Edge Case 5: Skip repositories where issues are disabled
+        # Edge Case 5: Repositories with Issues Disabled
         if not repo.has_issues:
-            continue  # Skip this repository
+            continue  # Skip repositories without issues enabled
 
-        # Edge Case 6: Skip archived repositories
+        # Edge Case 6: Archived or Disabled Repositories
         if repo.archived:
-            continue  # Skip this repository
+            continue  # Skip archived repositories
 
-        # Edge Case 21: Skip forked repositories
+        # Edge Case 21: Repository Forks
         if repo.fork:
-            continue  # Skip this repository
-
-        # Edge Case 8: Handle renamed repositories
-        repo_name = repo.name
+            continue  # Skip forked repositories
```

Comment:
Might the community quite appreciate understanding the logging nuances here? Namely, can we track precisely who's cloned the repo and what they're doing with it? Perhaps this could help us organically find contributors to add as new DUNA members? 💭

Reply:
I could be wrong, but I believe this skips forked repos so that only issues within the main repo are added to the Project. Once we get that up and running, perhaps we can create something similar but for forks and issues created in the forks?

Reply:
Is it possible to publicize when forks are created, and by which accounts? 

I suppose some folks might rather work quietly, and we could track and seek out good contributors based on comments or push requests rather than forks.

Reply:
I didn't realize this was visible to anyone! In that case I think we're fine here with this for the forseeable future.